[[ Bug 22868 ]] Ensure format() accepts negative length

livecodepanos opened the pull request:
This patch ensures the format() fuction recognizes a negative format length. In this case, it appends spaces to the string, matching the LC 6.x behavior.

Fixes https://quality.livecode.com/show_bug.cgi?id=22868

runrevmark commented on engine/src/exec-strings.cpp:
```diff
@@ -1227,9 +1232,19 @@ void MCStringsEvalFormat(MCExecContext& ctxt, MCStringRef p_format, MCValueRef*
+                                if (!t_is_negative)
+                                    t_success = MCStringAppendFormat(*t_result, "%0*s%@", width - t_range . length, "", *t_string);
+                                else
+                                    t_success = MCStringAppendFormat(*t_result, "%@%0*s", *t_string, width - t_range . length, "");
```
Is this definitely the LC6 behavior? i.e. if 0 padding is specified, and -ve (i.e. left alignment in width) then 0's appear after the number (thus changing the number printed...).

livecodepanos replied:
Indeed, in the case of 0 padding this will not match the LC 6 behavior. I have just pushed an update to fix this use-case too:

LC 6: `put 6 into n1;put format ("%-05s", n1)` --> `6`
LC 7+: `put 6 into n1;put format ("%-05s", n1)` --> `00006`
this patch: `put 6 into n1;put format ("%-05s", n1)` --> `6`

livecodepanos committed:
[[ Bug 22868 ]] Ensure format() accepts negative length

This patch ensures the format() fuction recognizes a negative format
length. In this case, it appends spaces to the string, matching the
LC 6.x behavior.

runrevmark commented:
@livecode-vulcan review ok 8f062ea

livecode-vulcan commented:
💙 review by @runrevmark ok 8f062ea9f99800354c371ac869420eafe80fec91

livecode-vulcan commented:
😎 test success 8f062ea9f99800354c371ac869420eafe80fec91

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/1043)
* try-community-armv7-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-ndk16r15/builds/451)
* try-community-arm64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-ndk16r15/builds/450)
* try-community-x86-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-ndk16r15/builds/449)
* try-community-x86_64-android-ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-ndk16r15/builds/450)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/2568)
* try-community-universal-ios-iphoneos14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.5/builds/45)
* try-community-universal-ios-iphoneos14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos14.4/builds/97)
* try-community-universal-ios-iphonesimulator14.5: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.5/builds/45)
* try-community-universal-ios-iphonesimulator14.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator14.4/builds/96)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/2594)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/2584)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/2582)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/2619)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/2555)
